micronaut-camunda-bpm-feature/src/test/kotlin/info/novatec/micronaut/camunda/bpm/feature/test/EventingPublisherPluginTest.kt

```diff
@@ -0,0 +1,145 @@
+
+    @MicronautTest
+    @Nested
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    inner class EventingTestPluginNotLoaded : TestPropertyProvider {
+
+        override fun getProperties(): MutableMap<String, String> {
+            return mutableMapOf()
+        }
+
+        @Inject
+        lateinit var processEngineConfiguration: MnProcessEngineConfiguration
+
+        @Inject
+        lateinit var processEngine: ProcessEngine
+
+        @Inject
+        lateinit var configuration: Configuration
+
+        @Inject
+        lateinit var eventListener: EventListener
+
+        @Test
+        fun `check that plugin is not loaded`() {
+            val plugins = processEngineConfiguration.processEnginePlugins
+            assertEquals(0, plugins.size)
+        }
+
+        @Test
+        fun `check no events are published`() {
+            assertTrue(eventListener.count == 0)
+        }
+    }
+
+    @MicronautTest
+    @Nested
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    inner class EventingTestPluginLoaded : TestPropertyProvider {
+
+        override fun getProperties(): MutableMap<String, String> {
+            return mutableMapOf(
+                    "camunda.eventing.enabled" to "true",
+                    "camunda.eventing.execution" to "true",
+                    "camunda.eventing.task" to "true",
+                    "camunda.eventing.history" to "true"
+            )
+        }
+
+        @Inject
+        lateinit var processEngineConfiguration: MnProcessEngineConfiguration
+
+        @Inject
+        lateinit var processEngine: ProcessEngine
+
+        @Inject
+        lateinit var configuration: Configuration
+
+        @Inject
+        lateinit var repositoryService: RepositoryService
+
+        @Inject
+        lateinit var runtimeService: RuntimeService
+
+        @Inject
+        lateinit var eventListener: EventListener
+
+        @Test
+        fun `check that plugin is loaded`() {
+            val plugins = processEngineConfiguration.processEnginePlugins
+            assertEquals(1, plugins.size)
+            assertTrue(plugins.get(0) is EventingPublisherPlugin)
+        }
+
+        @Test
+        fun `check all listeners are added`() {
+            // the first listener is the DefaultFailedJobParseListener, which is automatically added by ProcessEngineConfigurationImpl
+            assertEquals(2, processEngineConfiguration.customPostBPMNParseListeners.size)
+            assertEquals(1, processEngineConfiguration.customHistoryEventHandlers.size)
+        }
+
+        @Test
+        fun `check events are published`() {
+            val processId = "testProcess"
+            ProcessUtil.deploy(repositoryService, Bpmn.createProcess(processId)
+                    .executable()
+                    .startEvent()
+                    .userTask()
+                    .endEvent())
+            runtimeService.startProcessInstanceByKey(processId)
+
+            assertTrue(eventListener.count > 0)
```

Comment:
Actually, you could change all (?) asserts to use assertj, e.g. `assertThat((eventListener.count)).isGreaterThan(0)` The error messages are always nicer if it fails.